src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -204,69 +228,73 @@ async Task Process(BasicDeliverEventArgs message)
-
-                while (!processed && !errorHandled)
-                {
-                    try
-                    {
-                        var contextBag = new ContextBag();
-                        contextBag.Set(message);
-
-                        var messageContext = new MessageContext(messageId, headers, messageBody ?? Array.Empty<byte>(), TransportTransaction, contextBag);
+            var processed = false;
+            var errorHandled = false;
+            var numberOfDeliveryAttempts = 0;
+            var messageBody = message.Body.ToArray();
 
-                        await onMessage(messageContext, CancellationToken.None).ConfigureAwait(false);
-                        processed = true;
-                    }
-                    catch (Exception exception)
-                    {
-                        ++numberOfDeliveryAttempts;
-                        headers = messageConverter.RetrieveHeaders(message);
-                        var contextBag = new ContextBag();
-                        contextBag.Set(message);
+            while (!processed && !errorHandled)
+            {
+                var processingContext = new ContextBag();
 
-                        var errorContext = new ErrorContext(exception, headers, messageId, messageBody ?? Array.Empty<byte>(), TransportTransaction, numberOfDeliveryAttempts, contextBag);
+                processingContext.Set(message);
 
-                        try
-                        {
-                            errorHandled = await onError(errorContext, CancellationToken.None).ConfigureAwait(false) == ErrorHandleResult.Handled;
+                try
+                {
 
-                            if (!errorHandled)
-                            {
-                                headers = messageConverter.RetrieveHeaders(message);
-                            }
-                        }
-                        catch (Exception ex)
-                        {
-                            criticalErrorAction($"Failed to execute recoverability policy for message with native ID: `{messageId}`", ex, CancellationToken.None);
-                            await consumer.Model.BasicRejectAndRequeueIfOpen(message.DeliveryTag).ConfigureAwait(false);
+                    var messageContext = new MessageContext(messageId, headers, messageBody ?? Array.Empty<byte>(), TransportTransaction, processingContext);
 
-                            return;
-                        }
-                    }
+                    await onMessage(messageContext, messageProcessingCancellationTokenSource.Token).ConfigureAwait(false);
+                    processed = true;
                 }
-
-                if (processed && tokenSource.IsCancellationRequested)
+                catch (OperationCanceledException ex) when (messageProcessingCancellationTokenSource.IsCancellationRequested)
                 {
+                    Logger.Info("Message processing cancelled.", ex);
```

Comment:
Is it weird to do an Info that carries an exception? When the endpoint is shutting down do you really need a dozen exception traces stuffed into the log file? I don't think I've logged like this in any of the work I've done.

Reply:
I agree, should we remove the ex or skip logging here completely? (cc @adamralph )

Reply:
I'm clutching at straws here, but I guess it *might* be useful to look at the stack trace to see exactly where the operation was cancelled, but that is a `Debug()` thing at best.

However, it probably makes sense to keep the `Info()` (we've done that in other transports), and I'm not sure we'd ever go so far as doing this:

```suggestion
                    Logger.Info("Message processing cancelled.");
                    Logger.Debug("Message processing cancelled.", ex);
```

Reply:
Discussed with @andreasohlund and we agreed to remove the logging for now, to align with changes already merged in other transports.